strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/AuthorizationConfigController.java

```diff
@@ -0,0 +1,245 @@
+                            @ApiResponse(code = 500, message = "An error occurred.") })
+    @RequestMapping(value = "/xml", method = RequestMethod.GET, produces = { MediaType.APPLICATION_XML_VALUE,
+                                                                             MediaType.APPLICATION_JSON_VALUE })
+    public synchronized ResponseEntity getAuthorizationConfig()
+            throws JAXBException
+    {
+        logger.debug("Trying to receive authorization config as XML file...");
+
+        // TODO Due to internal error in spring-data-orientdb
+        // com.orientechnologies.orient.client.remote.OStorageRemote cannot be cast to
+        // com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage
+        return processConfig(null, config -> ResponseEntity.ok(configGenericParser.serialize(config)));
+    }
+
+    // ----------------------------------------------------------------------------------------------------------------
+    // Revoke role by name
+
+    @ApiOperation(value = "Deletes a role by name.", position = 3)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The role was deleted."),
+                            @ApiResponse(code = 400, message = "Bad request.")
+    })
+    @RequestMapping(value = "role/{name}", method = RequestMethod.DELETE)
+    public ResponseEntity deleteRole(@ApiParam(value = "The name of the role", required = true)
+                                     @PathVariable("name") String name)
+            throws Exception
+    {
+        return processConfig(config ->
+                             {
+
+                                 // find Privilege by name
+                                 Role target = null;
+                                 for (Role role : config.getRoles().getRoles())
+                                 {
+                                     if (role.getName().equalsIgnoreCase(name))
+                                     {
+                                         target = role;
+                                         break;
+                                     }
+                                 }
+                                 if (target != null)
+                                 {
+                                     // revoke role from current config
+                                     config.getRoles().getRoles().remove(target);
+                                     configProvider.updateConfig(config);
+
+                                     // revoke role from every user that exists in the system
+                                     getAllUsers().forEach(user ->
+                                                           {
+                                                               if (user.getRoles().remove(name.toUpperCase()))
+                                                               {
+                                                                   // evict such kind of users from cache
```

Comment:
I'm not following something here... What kind of users are these? Would you mind explaining how this works?